source/text/typesetter.ts

```diff
@@ -226,84 +307,192 @@ export class Typesetter {
         }
     }
 
+    /**
+     * Adjusts the vertices for a line after typesetting (done due to line feed, word wrap, or end of line) w.r.t.
+     * the targeted line alignment.
+     * @param width - Width of the line (e.g., typesetting position at the end of the line in typesetting space).
+     * @param alignment - Targeted alignment, e.g., left, center, or right.
+     * @param vertices - Glyph vertices for rendering to align the origins' x-components of (expected untransformed).
+     * @param begin - Vertex index to start alignment at.
+     * @param end - Vertex index to stop alignment at.
+     */
+    private static transformAlignment(width: number, alignment: Label.Alignment,
+        vertices: GlyphVertices | undefined, begin: number, end: number): void {
+        if (vertices === undefined || alignment === Label.Alignment.Left) {
+            return;
+        }
+
+        let offset = -width;
+        if (alignment === Label.Alignment.Center) {
+            offset *= 0.5;
+        }
+
+        /* Origin is expected to be in typesetting space (not transformed yet). */
+        for (let i = begin; i < end; ++i) {
+            vertices.vertices[i].origin[0] += offset;
+        }
+    }
 
     /**
-     * Transforms the labels extent (typesetting space to, e.g., world space or screen space).
-     * @param transform - Transformation that was applied to every vertex.
-     * @param extent - Untransformed label extent (in typesetting space) to be transformed.
-     * @returns - Transformed label extent (copy).
+     * Create and transform glyph vertices for rendering.
```

Comment:
doc: where does the *created* vertices go? What exactly do we *transform* here?
Furthermore, it helps to document when a param is a in/out param.